src/compiler/watchUtilities.ts

```diff
@@ -120,23 +141,10 @@ namespace ts {
             }
         }
 
-        function fileNameEqual(name1: string, name2: string) {
-            return getCanonicalFileName(name1) === getCanonicalFileName(name2);
-        }
-
-        function hasEntry(entries: readonly string[], name: string) {
-            return some(entries, file => fileNameEqual(file, name));
-        }
-
-        function updateFileSystemEntry(entries: string[], baseName: string, isValid: boolean) {
-            if (hasEntry(entries, baseName)) {
-                if (!isValid) {
-                    return filterMutate(entries, entry => !fileNameEqual(entry, baseName));
-                }
-            }
-            else if (isValid) {
-                return entries.push(baseName);
-            }
+        function hasEntry(entries: readonly Canonicalized[], name: Canonicalized) {

```

Comment:
Maybe `sortedEntries` to make it clear that these entries must be sorted?

Reply:
I'll use `SortedArray` and `SortedReadonlyArray` instead.